mediaire_toolbox/transaction_db/model.py

```diff
@@ -64,20 +66,89 @@ def to_dict(self):
                  'sequences': self.sequences,
                  'archived': self.archived,
                  'patient_consent': self.patient_consent
-                 }
+                }
 
     def __repr__(self):
         return "<Transaction(transaction_id='%s', patient_id='%s', start_date='%s')>" % (
             self.transaction_id, self.patient_id, self.start_date)
 
 
+class User(Base):
+    
+    """for multi-tenant pipelines, users might be required"""
+    __tablename__ = 'users'
+        
+    id = Column(Integer, Sequence('id'), primary_key=True)
+    name = Column(String(255), unique=True)
+    hashed_password = Column(String(128))
+    added = Column(DateTime(), default=datetime.datetime.utcnow)
+    
+    @staticmethod
+    def password_hash(password):
+        return pwd_context.hash(password)
+
+    def verify_password(self, password):
+        return pwd_context.verify(password, self.hashed_password)
+    
+    def to_dict(self):
+        return { 'id': self.id,
+                 'name': self.name,
+                 'hashed_password': self.hashed_password,
+                 'added': self.added.strftime("%Y-%m-%d %H:%M:%S") }
+        
+
+class UserTransaction(Base):
+    
+    """for multi-tenant pipelines, transactions might be associated with users"""
+    __tablename__ = 'users_transactions'
+        
+    user_id = Column(Integer, ForeignKey('users.id'),
+                     primary_key=True)
+    transaction_id = Column(Integer, ForeignKey('transactions.transaction_id'),
+                            primary_key=True)
+    
+    def to_dict(self):
+        return { 'user_id': self.user_id,
+                 'transaction_id': self.transaction_id }
+
+
+class UserRole(Base):
+    
+    """for multi-tenant pipelines, users might have different roles in the 
+    associated platform"""
+    __tablename__ = 'users_roles'
+        
+    user_id = Column(Integer, ForeignKey('users.id'),
+                     primary_key=True)
+    role_id = Column(String(64), ForeignKey('roles.role_id'),
+                     primary_key=True)
+    
+    def to_dict(self):
+        return { 'user_id': self.user_id,
+                 'role_id': self.role_id }
+    
+    
+class Role(Base):
```

Comment:
I think we need to add a permission attribute that indicates what permissions an added role has. Otherwise we need to hardcode what permission a certain role id has.